### PR TITLE
SALTO-3721: exclude team managed boards and handle team managed projects

### PR DESCRIPTION
**Pull request by IdoZakk**
Preventing the fetch of team managed boards, and blocking deployment of team managed projects

---

_Additional context for reviewer_

---
_Release Notes_: 
Jira Adapter: 
* Team-managed boards will not be fetched
* Team-managed projects will not be deployable

---
_User Notifications_: 
Jira Adapter:
* Team-managed boards will be removed from users' environments

### CODE REVIEW

File: packages/jira-adapter/test/change_validators/projects/team_managed_project.test.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    projectInstance.value.style = 'classic'
+    expect(
+      await changeValidator([
+        toChange({
+          after: projectInstance,
+        }),
+      ]),
+    ).toEqual([])
+
+    projectInstance.value.projectCategory = undefined
```
Comment (shayc331) [inline packages/jira-adapter/test/change_validators/projects/team_managed_project.test.ts:77]: why is this needed?

File: packages/jira-adapter/src/client/pagination.ts [hunk 2]
```diff
@@ -25,16 +25,18 @@ const ITEM_INDEX_PAGINATION_URLS = [
...
 ]
 
+// filters out entries of specific project scope as we don't support it,
+// but leaves entries of global scope
+const notTeamScopeObject = (obj: clientUtils.ResponseValue): boolean =>
+  !(_.isPlainObject(obj) && 'scope' in obj && !_.isEqual(obj.scope, { type: 'GLOBAL' }))
+
+// filters out boards located in team managed projects
+const notTeamBoard = (obj: clientUtils.ResponseValue): boolean =>
+  !(_.isPlainObject(obj) && 'type' in obj && obj.type === 'simple')
```
Comment (shayc331) [inline packages/jira-adapter/src/client/pagination.ts:35]: I'm afraid this condition may be true for other cases as well, is there a condition we can add to verify it is a `board` object?

File: packages/jira-adapter/test/change_validators/projects/team_managed_project.test.ts [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    ).toEqual([
+      {
+        elemID: projectInstance.elemID,
+        severity: 'Error',
+        message: "Can't deploy a team-managed project",
+        detailedMessage: 'Currently team-managed projects are not supported. The project will not be deployed.',
+      },
+    ])
+  })
+  it('should do nothing a company managed project', async () => {
```
Comment (shayc331) [inline packages/jira-adapter/test/change_validators/projects/team_managed_project.test.ts:67]:
```suggestion
  it('should do nothing when changing a company managed project', async () => {
```

File: packages/jira-adapter/src/client/pagination.ts [hunk 4]
```diff
@@ -25,16 +25,18 @@ const ITEM_INDEX_PAGINATION_URLS = [
...
   if (Array.isArray(response)) {
-    return response
-      .filter(
-        item =>
-          // filters out entries of specific project scope as we don't support it,
-          // but leaves entries of global scope
-          !(_.isPlainObject(item) && 'scope' in item && !_.isEqual(item.scope, { type: 'GLOBAL' })),
-      )
-      .flatMap(removeScopedObjectsImpl) as T[]
+    return response.filter(notTeamScopeObject).filter(notTeamBoard).flatMap(removeScopedObjectsImpl) as T[]
```
Comment (shayc331) [inline packages/jira-adapter/src/client/pagination.ts:39]: very nice prettier 💯

**Review by shayc331 — APPROVED** [review]
amazing 🏆 
We will need noise reduction as well since we wont fetch team managed boards any longer, so please merge it after the noise reduction is inside

File: packages/jira-adapter/src/client/pagination.ts (hunk 2 above)
Comment (IdoZakk) [reply packages/jira-adapter/src/client/pagination.ts:35]: It is a bit tricky, but I added it using the "self" field

File: packages/jira-adapter/src/change_validators/projects/team_managed_project.ts [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    .filter(isInstanceChange)
+    .filter(isAdditionOrModificationChange)
+    .filter(change => change.data.after.elemID.typeName === PROJECT_TYPE)
+    .map(getChangeData)
+    .filter(instance => instance.value.style === 'next-gen')
+    .map(instance => ({
+      elemID: instance.elemID,
+      severity: 'Error' as SeverityLevel,
+      message: "Can't deploy a team-managed project",
+      detailedMessage: 'Currently team-managed projects are not supported. The project will not be deployed.',
```
[the remark this answers is not in the document]
Comment (IdoZakk) [reply packages/jira-adapter/src/change_validators/projects/team_managed_project.ts:42]: Yes
